File: Java-Core/src/main/java/presto/error/UserError.java
package presto.error;

import presto.expression.IExpression;
import presto.runtime.Context;


public class UserError extends ExecutionError {

	private static final long serialVersionUID = 1L;

	IExpression expression;
	
	public UserError(IExpression expression) {
		this.expression = expression;
	}
	
	@Override
	public IExpression getExpression(Context context) {
		return expression;
	}
}

File: Java-CodeServer/src/main/resources/js/prompto/error/UserError.js
var ExecutionError = require("./ExecutionError").ExecutionError;

function UserError(expression) {
	ExecutionError.call(this);
	this.expression = expression;
	return this;
}

UserError.prototype = Object.create(ExecutionError.prototype);
UserError.prototype.constructor = UserError;

UserError.prototype.getExpression = function(context) {
	return this.expression;
};

exports.UserError = UserError;
